omnigibson/utils/geometry_utils.py

```diff
@@ -238,7 +238,7 @@ def _generate_convex_hull_volume_checker_functions(convex_hull_mesh):
                 USD mesh
     """
     # For efficiency, we pre-compute the mesh using trimesh and find its corresponding faces and normals
-    trimesh_mesh = mesh_prim_to_trimesh_mesh(convex_hull_mesh, include_normals=False, include_texcoord=False).convex_hull
+    trimesh_mesh = mesh_prim_mesh_to_trimesh_mesh(convex_hull_mesh, include_normals=False, include_texcoord=False).convex_hull
```

Comment:
I'm getting a feeling that a lot of stuff in this function (the pos-quat-scale stuff especially) can possibly have issues with the skew etc right? Should we try to unify everything to use the 4x4 scaled transform matrices instead of these adhoc partial transforms?

Reply:
I don't think this has issues with the skew. The checker here is to transform points from the link frame to the mesh frame, so the math is accurate (pos, quat, scale should be enough). I agree it's possible to unify everything using the scaled transform matrices, but this part of the code is kind of complicated and well-tested and I don't want to break it. We can unify later I think.